docs/src/ref_impl/build_and_run.md

```diff
@@ -0,0 +1,104 @@
+
+Our supported targets are the following:
+
+| Hardware         | Architecture     | Scope         |
+|---               |---               |---            |
+| Generic Intel    | x86              | QEMU (VM)     |
+| NVIDIA Jetson AGX Orin  | AArch64   | hardware      |
+| NXP i.MX 8QM-MEK | AArch64          | hardware      |
+
+Scope of target support is updated with development progress.
+
+## Building Ghaf Image
+
+Before you begin, follow the basic device-independent steps:
+
+* Clone the git repository <https://github.com/tiiuae/ghaf>.
+* Ghaf uses a Nix flake approach to build the framework targets, make sure to:
+  * Install Nix or full NixOS if needed: <https://nixos.org/download.html>.
+  * Enable flakes: <https://nixos.wiki/wiki/Flakes#Enable_flakes>.  
+    To see all Ghaf supported outputs, type `nix flake show`.
+  * Set up an AArch64 remote builder: <https://nixos.org/manual/nix/stable/advanced-topics/distributed-builds.html>.
+
+---
+
+
+### Running Ghaf Image for x86 VM (ghaf-host)
+
+Run the `nix run .#packages.x86_64-linux.vm` command.
+
+This creates `ghaf-host.qcow2` copy-on-write overlay disk image in your current directory. If you do unclean shutdown for the QEMU VM, you might get weird errors the next time you boot. Simply removing `ghaf-host.qcow2` should be enough. To cleanly shut down the VM, from the menu bar of the QEMU Window, click Machine and then Power Down.
+
+
+---
+
+### Building Ghaf Image for NVIDIA Jetson Orin AGX
+
+If you cloned the ghaf repository and installed an AArch64 remote builder, run the `nix build .#packages.aarch64-linux.nvidia-jetson-orin-release` command to build the release build.
+
+Otherwise, you can use the `nix build github:tiiuae/ghaf#nvidia-jetson-orin-debug-flash-script` command to run a script that builds your image without any additional installations. Give "yes" answers to all script questions. The building process takes around 1,5 hour.
+
+In addition, if a [cross-compilation](https://tiiuae.github.io/ghaf/build_config/cross_compilation.html)support is needed, enable `binfmt` in your `configuration.nix`:
+
+    ```
+    boot.binfmt.emulatedSystems = [
+      "aarch64-linux"
+    ];
+    ```
+After, you can prepare a USB boot media with the target image you built: `dd if=./result/nixos.img of=/dev/<YOUR_USB_DRIVE> bs=32M`. Boot the hardware from the USB media.
+
+
+#### Flashing NVIDIA Jetson Orin AGX
+
+1. Set up the following connections:
+   1. Connect the board to a power supply with a USB-C cable.
+   2. Connect a Linux laptop to the board with the USB-C cable.
+   3. Connect the Linux laptop to the board with a Micro-USB cable.
+   > For more information on the board's connections details, see the [Hardware Layout](https://developer.nvidia.com/embedded/learn/jetson-agx-orin-devkit-user-guide/developer_kit_layout.html) section of the Jetson AGX Orin Developer Kit User Guide.
```

Comment:
This could benefit from diagram or image on how to connect the cables. For anyone to quick start.